Makefile

```diff
@@ -306,8 +306,9 @@ tools/bin/ineffassign:tools/check/go.mod
 	cd tools/check; \
 	$(GO) build -o ../bin/ineffassign github.com/gordonklaus/ineffassign
 
-tools/bin/errdoc-gen: go.mod
-	$(GO) build -o $@ github.com/pingcap/tiup/components/errdoc/errdoc-gen
```

Comment:
This way the `$(GO) build` operation would pollute the `go.mod` file, introducing unnecessary dependency or even change some package's version.